pkg/config/config.go

```diff
@@ -271,3 +276,33 @@ func (cfg *Config) addPlugin(plugin plugin.Interface) {
 func (cfg *Config) getPlugins() []plugin.Interface {
 	return cfg.LoadedPlugins
 }
+
+// Merge src into the cfg config only overwriting zero values in cfg.
+func (cfg *Config) Merge(src *Config) {
+	// if the search path is provided use that directly
+	customSearchPath := make([]string, len(cfg.PluginSearchPath))
+	if len(cfg.PluginSearchPath) > 0 {
+		copy(customSearchPath, cfg.PluginSearchPath)
+	}
+
+	mergo.Merge(cfg, src)
+
+	if len(customSearchPath) > 0 {
+		cfg.PluginSearchPath = customSearchPath
+	}
+
+	// TODO(chuckha) here is a safety net in case a supplied plugin search path is provided in the custom config.
+	// mergo will append the defaults to the provided slice which may not be the behavior we want.
+	// This could go away if we change the assumptions about loading plugins change.
+	uniqueSearchPaths := make(map[string]struct{})
+	for _, searchPath := range cfg.PluginSearchPath {
+		if _, ok := uniqueSearchPaths[searchPath]; !ok {
```

Comment:
why not just unconditionally insert the element? should be an effective no-op if you're inserting the same thing

Reply:
also `bool` might make it more obvious you're just using this as a ersatz set

Reply:
empty struct is convention for a map used as a set, at least from what i've seen. I think kubernetes straight up maps `type Empty struct{}` for the sets it uses. Either way this code is going away

Reply:
huh, TIL